torch/onnx/symbolic_opset9.py

```diff
@@ -1253,7 +1258,7 @@ def _generic_rnn(g, variant, input, initial_states, all_weights, has_biases,
         reform_permutation = [(0, 1), (3, 4), (1, 3)]
 
     def reform_weights(g, w, n, intervals):
-        slices = [g.op('Slice', w, axes_i=[0], starts_i=[x * n], ends_i=[y * n]) for x, y in intervals]
+        slices = [_slice_op(g, w, axes=[0], starts=[x * n], ends=[y * n]) for x, y in intervals]
```

Comment:
Exporting a v10 slice in v9 model is incorrect behavior.

Reply:
That is not the case.

Calling _slice_op() will redirect the call to symbolic_opset10._slice_op() if the opset_version is 10, or execute for opset 9 if the opset_version is 9.

The idea is to replace all occurrences to g.op('Slice') by _slice_op(), that will take care of executing the right code.

When reform_weights is called with opset 10, since it is not overwritten for opset 10, we will fallback to symbolic_opset9.reform_weights. 
symbolic_opset9.reform_weights uses Slice that is implemented for opset 9 and overwritten for opset 10. So if the opset we are dealing with is 10, we should use the one in opset 10; the code in _slice_op() deals with this scenario. 

So without this change, we would find opset9 slice ops in opset 10 models.
(initially, we should have had reform_weights and retrieve_state in the blacklist, when slice was in the blacklist)